src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -104,6 +120,19 @@
 
     private @Nonnull String excludes = DescriptorImpl.defaultExcludes;
 
+    /** Whether to build regular origin branches. */
+    private @Nonnull Boolean buildOriginBranch = DescriptorImpl.defaultBuildOriginBranch;
+    /** Whether to build origin branches which happen to also have a PR filed from them (but here we are naming and building as a branch). */
+    private @Nonnull Boolean buildOriginBranchWithPR = DescriptorImpl.defaultBuildOriginBranchWithPR;
+    /** Whether to build PRs filed from the origin, where the build is of the merge with the base branch. */
+    private @Nonnull Boolean buildOriginPRMerge = DescriptorImpl.defaultBuildOriginPRMerge;
+    /** Whether to build PRs filed from the origin, where the build is of the branch head. */
+    private @Nonnull Boolean buildOriginPRHead = DescriptorImpl.defaultBuildOriginPRHead;
+    /** Whether to build PRs filed from a fork, where the build is of the merge with the base branch. */
+    private @Nonnull Boolean buildForkPRMerge = DescriptorImpl.defaultBuildForkPRMerge;
+    /** Whether to build PRs filed from a fork, where the build is of the branch head. */
+    private @Nonnull Boolean buildForkPRHead = DescriptorImpl.defaultBuildForkPRHead;
```

Comment:
NIT ... I'm after seeing these `build*` flags together in a few places in the PR. Seems like they are all closely related and could be encapsulated into a type of some sort.